### DISCUSSION TITLE
No updates for hardened?

### DISCUSSION THREAD

**Discussion by fdw** [stilliard/docker-pure-ftpd#163] [Q&A]
According to the readme, `hardened` is based on `latest` with some configuration changes. However, the `hardened` image is currently eight months old, whereas `latest` regularly sees some changes.

Does it still make sense to use the `hardened` image?

**Comment by stilliard** [comment]
Hey @fdw 
Thanks for raising this, it's a good point, I'm wondering if it makes more sense to move the hardened changes into the main branch just as a flag. 

The only difference between the 2 branches atm is the flags we pass to pure-ftpd are slightly more restrictive by default in hardened but I no longer recommend this way by default as these restrictions were often the cause of many of the issues reported here. 

You can see it's differences here on the files changed tab:
https://github.com/stilliard/docker-pure-ftpd/compare/master...hardened

Switching that compare url around we can see what's missing in hardened that's been recently added to the main branch:
https://github.com/stilliard/docker-pure-ftpd/compare/hardened...master

It's only a couple changes missing, no big updates lately, I'll merge these to hardened shortly to keep it up to date for now though.

**Reply by stilliard** [reply]
Quick update there, pushed the latest few changes from the main branch to the hardened one now too. 

Still, switching from a different branch to just a flag such as `--hardened` you can pass at runtime does seem like a better long term solution to make sure both are up to date :)

**Reply by fdw** [reply]
Thanks for the quick and extensive answer! I agree that mainting both independently is more effort than ideal. Having just a flag would be fine for me.

**Reply by stilliard** [reply]
No problem, thanks for your feedback on this. 
I've added an issue in case anyone wants to work on a PR for it, or I'll work on it some time soon otherwise.

https://github.com/stilliard/docker-pure-ftpd/issues/164
